package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -205,10 +204,8 @@ function M.generate_mac(i)
 	-- a hardware MAC filter. (e.g 'rt305x')
 
 	m6 = bit.band(m6, 0xF8) -- zero the last three bits (space needed for counting)
-	m6 = m6 + i                   -- add virtual interface id
+	m6 = math.fmod(m6 + i, 256)   -- add virtual interface id
```

Comment:
This commit needs to be squashed into the previous.

Instead of `fmod`, it's better to use `m6 = bit.band(m6 + i, 0xFF)` for consistency with the rest of the code.

Reply:
Thanks!
Just a question: why is it needed to "zero the last three bits (space needed for counting)"?
I think this is just convenient as one knows which MAC is which, but it would not be required if I am not mistaken..?

Reply:
For some radios, all VIFs need to use MAC addresses that only differ in the last 3 bits. This was achieved by setting the last 3 bits to 0 and counting up for up to 8 addresses.

Reply:
Thanks, then I will leave it as it is now